src/compas_model/model/model.py

```diff
@@ -108,6 +108,7 @@ def __init__(self, name=None):
         self._graph = InteractionGraph()
         self._graph.update_default_node_attributes(element=None)
         self._elementlist = []
+        self._materials = []
```

Comment:
assuming that material keys are used to identify materials, i would make this a dict...